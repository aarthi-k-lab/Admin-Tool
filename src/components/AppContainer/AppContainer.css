.app {
  display: flex;
  flex-direction: column;
  height: 100%;
}

:global(.material-ui-button.material-ui-button) {
  min-height: 2.25rem;
  min-width: 4rem;
}
