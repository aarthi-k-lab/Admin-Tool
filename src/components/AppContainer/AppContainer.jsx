import React from 'react';
import PropTypes from 'prop-types';

import './AppContainer.css';
import Footer from 'components/Footer';

const AppContainer = ({ children, hideFooter }) => (
  <div styleName="app">
    { children }
    { hideFooter ? null : <Footer /> }
  </div>
);

AppContainer.defaultProps = {
  hideFooter: false,
};

AppContainer.propTypes = {
  children: PropTypes.node.isRequired,
  hideFooter: PropTypes.bool,
};

export default AppContainer;
