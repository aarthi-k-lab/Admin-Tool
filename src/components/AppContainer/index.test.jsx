import React from 'react';
import renderer from 'react-test-renderer';
import AppContainer from '.';

describe('<AppContainer />', () => {
  test('snapshot test with footer', () => {
    const tree = renderer
      .create(<AppContainer>content</AppContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('snapshot test without footer', () => {
    const tree = renderer
      .create(<AppContainer hideFooter>content</AppContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
