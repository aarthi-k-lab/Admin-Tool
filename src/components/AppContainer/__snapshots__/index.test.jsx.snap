// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer /> snapshot test with footer 1`] = `
<div
  className="components-AppContainer-AppContainer__app--_pvE9"
>
  content
  <footer
    className="components-Footer-Footer__footer--2oWe5"
  >
    Â© Nationstar Mortgage LLC, All rights reserved
  </footer>
</div>
`;

exports[`<AppContainer /> snapshot test without footer 1`] = `
<div
  className="components-AppContainer-AppContainer__app--_pvE9"
>
  content
</div>
`;
